x-pack/plugins/apm/server/routes/suggestions/get_suggestions_by_service_name.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ProcessorEvent } from '../../../common/processor_event';
+import { SERVICE_NAME } from '../../../common/elasticsearch_fieldnames';
+import { rangeQuery, termQuery } from '../../../../observability/server';
+import { Setup } from '../../lib/helpers/setup_request';
+
+export async function getSuggestionsByServiceName({
+  fieldName,
+  fieldValue,
+  serviceName,
+  setup,
+  size,
+  start,
+  end,
+}: {
+  fieldName: string;
+  fieldValue: string;
+  serviceName: string;
+  setup: Setup;
+  size: number;
+  start?: number;
+  end?: number;
+}) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search(
+    'get_suggestions_by_service_name',
+    {
+      apm: {
+        events: [
+          ProcessorEvent.transaction,
+          ProcessorEvent.error,
+          ProcessorEvent.metric,
+        ],
+      },
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...termQuery(SERVICE_NAME, serviceName),
+              ...(start && end ? rangeQuery(start, end) : []),
```

Comment:
I think it's best that we always have a range set - otherwise this will look at _all_ the data which could take a while. If we don't have a time picker I'd suggest something like `now-24h`.

Reply:
good point here 👍 

I set the default ranges to `now-24h` 

https://github.com/elastic/kibana/pull/126679/commits/bb56b5e256286d4dbd62f4cf17ab40352a26252d